docs/guides/component-testing/component-framework-configuration.mdx

```diff
@@ -45,26 +70,105 @@ resolves to a `devServer` instance.
 
-## Custom Index File
+Any requests triggered during a test using the `devServerPublicPathRoute` as
+defined in the `cypressConfig` will be forwarded to your server. Cypress will
+trigger a request for `[devServerPublicPathRoute]/index.html` when a test is
+started. Your server needs to reply with the html-file referenced in
+`cypressConfig.indexHtmlFile` and inject a script to load the support files and
+the actual test.
 
-By default, Cypress renders your components into an HTML file located at
-`cypress/support/component-index.html`.
+```ts
+function createServer(cypressConfig, bundleDir, port = 1234) {
+  const app = express()
+
+  // read kickstart script - see below for an example
+  const clientScript = readFileSync(
+    path.join(__dirname, './client-script.js'),
+    'utf8'
+  )
+
+  app.get(
+    cypressConfig.devServerPublicPathRoute + '/index.html',
+    async (_req, res) => {
+      // read custom index.html file
+      const html = await fs.readFile(
+        path.join(cypressConfig.repoRoot, cypressConfig.indexHtmlFile),
+        { encoding: 'utf8' }
+      )
+
+      // inject kickstart-script
+      const output = html.replace(
+        '</head>',
+        `<script type="module">${clientScript}</script></head>`
+      )
+      res.send(output)
+    }
+  )
+
+  // you need to establish some url-to-path-mapping, if your bundler outputs
+  // the full directory structure you can map this one to one
+  app.use(cypressConfig.devServerPublicPathRoute, express.static(bundleDir))
+
+  app.listen(port)
+}
+```
 
-The index file allows you to add in global assets, such as styles, fonts, and
-external scripts.
+For a real-world example, you can refer to
+[this loader](https://github.com/cypress-io/cypress/blob/466155c2125476374d9f9549530f67d0c6354a41/npm/vite-dev-server/src/plugins/cypress.ts#L82-L92)
+used by the vite-devserver.
 
-You can provide an alternative path to the file using the `indexHtmlFile` option
-in the [component config](/guides/references/configuration#component) options:
+The client script must retrieve information on the currently active test from
+the Cypress instance of the parent frame and load the corresponding bundle. If a
+support file is defined, it should be injected at the top of your test bundle or
+loaded before the test script.
 
-```js
-{
-  component: {
-    devServer,
-    indexHtmlFile: '/custom/path/to/component-index.html'
-  }
+```ts
+const CypressInstance = (window.Cypress = parent.Cypress)
+const devServerPublicPathRoute = CypressInstance.config(
+  'devServerPublicPathRoute'
+)
+
+const imports = []
+
+// If you do not bundle your support file along with the tests,
+// you need to add a separate import statement for the support file.
+if (CypressInstance.config('supportFile')) {
+  const absolute = CypressInstance.config('supportFile')
+  const relative = absolute.replace(CypressInstance.config('projectRoot'), '')
+  const relativeUrl = `${devServerPublicPathRoute}${relative}`
+  imports.push({
```

Comment:
I had a closer look into the load mechanism, and this format causes the script to be loaded twice:
https://github.com/cypress-io/cypress/blob/7a33f5c1a87f5a3aa3b8da575b39e42f93bfd5db/packages/driver/src/cypress/script_utils.ts#L6
First it will be fetched in the `fetchScript` function, and later in `evalScripts` the load-function is triggered, wich causes the import to fetch the contents again.

It's propably better to just pass the importFunctions and use the alternative path here: https://github.com/cypress-io/cypress/blob/7a33f5c1a87f5a3aa3b8da575b39e42f93bfd5db/packages/driver/src/cypress/script_utils.ts#L49

Reply:
This propably should be fixed for the vite-dev-server as well